Committed for glibc 2.33

commit ce625044886da0a6e348f4def83f746e28587a0c
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Wed Aug 5 23:46:14 2020 +0200

    hurd: Implement basic sched_get/setscheduler
    
    * sysdeps/mach/hurd/sched_gets.c: New file.
    * sysdeps/mach/hurd/sched_sets.c: New file.

diff --git a/sysdeps/mach/hurd/sched_gets.c b/sysdeps/mach/hurd/sched_gets.c
new file mode 100644
index 0000000000..9fe31ce79e
--- /dev/null
+++ b/sysdeps/mach/hurd/sched_gets.c
@@ -0,0 +1,30 @@
+/* Copyright (C) 2020 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <https://www.gnu.org/licenses/>.  */
+
+#include <errno.h>
+#include <sched.h>
+#include <sys/types.h>
+#include <hurd.h>
+
+
+/* Retrieve scheduling algorithm for a particular purpose.  */
+int
+__sched_getscheduler (pid_t pid)
+{
+  return SCHED_OTHER;
+}
+weak_alias (__sched_getscheduler, sched_getscheduler)
diff --git a/sysdeps/mach/hurd/sched_sets.c b/sysdeps/mach/hurd/sched_sets.c
new file mode 100644
index 0000000000..54d368abf5
--- /dev/null
+++ b/sysdeps/mach/hurd/sched_sets.c
@@ -0,0 +1,45 @@
+/* Copyright (C) 1996-2020 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <https://www.gnu.org/licenses/>.  */
+
+#include <errno.h>
+#include <sched.h>
+#include <sys/types.h>
+#include <hurd.h>
+
+
+/* Set scheduling algorithm and/or parameters for a process.  */
+int
+__sched_setscheduler (pid_t pid, int policy, const struct sched_param *param)
+{
+  switch (policy)
+    {
+    case SCHED_OTHER:
+      if (param->sched_priority != 0)
+	return __hurd_fail (EINVAL);
+      break;
+
+    case SCHED_FIFO:
+    case SCHED_RR:
+      return __hurd_fail (ENOTSUP);
+
+    default:
+      return EINVAL;
+    }
+  return 0;
+}
+libc_hidden_def (__sched_setscheduler)
+weak_alias (__sched_setscheduler, sched_setscheduler)
